src/bl.cpp

```diff
@@ -507,29 +507,31 @@ static https_request_err_e downloadAndShow(const char *url)
             Log.info("%s [%d]: Content size: %d\r\n", __FILE__, __LINE__, size);
             Log.info("%s [%d]: Free heap size: %d\r\n", __FILE__, __LINE__, ESP.getFreeHeap());
             Log.info("%s [%d]: Payload - %s\r\n", __FILE__, __LINE__, payload.c_str());
-            JsonDocument doc;
-            DeserializationError error = deserializeJson(doc, payload);
+
+            auto apiResponse = parseResponse_apiDisplay(payload);
+            bool error = apiResponse.outcome == ApiDisplayOutcome::DeserializationError;
+
             if (error)
             {
               result = HTTPS_JSON_PARSING_ERR;
             }
             else if (!error && special_function == SF_NONE)
             {
-              uint64_t request_status = doc["status"];
+              uint64_t request_status = apiResponse.status;
               Log.info("%s [%d]: status: %d\r\n", __FILE__, __LINE__, request_status);
               switch (request_status)
               {
               case 0:
               {
-                String image_url = doc["image_url"];
-                update_firmware = doc["update_firmware"];
-                String firmware_url = doc["firmware_url"];
-                uint64_t rate = doc["refresh_rate"];
-                reset_firmware = doc["reset_firmware"];
+                String image_url = apiResponse.image_url;
```

Comment:
We could clean this up further, but for now I tried to minimize the diff to make it easier to review, and less likely to break anything.